etc/scripts/mongodb_skillsim_lookup.py

```diff
@@ -107,47 +198,183 @@ def clone_collection(self, src_mongodb_uri, src_database,src_collection):
+  bb_sanity = bb_parser.add_argument_group("Sanity checks to avoid faulty entries")
+  bb_sanity.add_argument(
+      '--lower-bound', '-l',
+      type=float,
+      default=0,
+      help='ignore entries with duration smaller than this')
+  bb_sanity.add_argument(
+      '--upper-bound', '-u',
+      type=float,
+      default=float('inf'),
+      help='ignore entries with duration smaller than this')
+  bb_log = bb_parser.add_argument_group("Blackboard log information")
+  bb_log.add_argument(
+      '--src-uri',
+      type=str,
+      help='The MongoDB URI of the blackboard log connection (default: %(default)s)',
       default='mongodb://localhost:27017/')
-  parser.add_argument(
+  bb_log.add_argument(
       '--src-db',
       type=str,
-      help='The name of the blackboard log database',
+      help='The name of the blackboard log database (default: %(default)s)',
       default='fflog')
-  parser.add_argument(
-      '--dst-db',
-      type=str,
-      help='The name of the skiller simulator lookup database',
-      default='skills')
-  parser.add_argument(
+  bb_log.add_argument(
       '--src-col',
       type=str,
-      help='The name of the blackboard log collection',
+      help='The name of the blackboard log collection (default: %(default)s)',
       default='SkillerInterface.Skiller')
-  parser.add_argument(
-      '--dst-col',
-      type=str,
-      help='The name of the skiller simulator lookup collection',
-      default='exec_times')
-  parser.add_argument(
+  bb_log.add_argument(
       '--drop-src-col',
       type=bool,
       help='Delete the skiller blackboard log collection afterwards',
       default=False)
-  args = parser.parse_args()
-  transformer = MongoTransformer(args.dst_uri,args.dst_db,args.dst_col)
-  transformer.transform(args.src_uri,args.src_db,args.src_col)
-  if args.drop_src_col:
-      drop_collection(args.src_mongodb_uri, args.src_db, args.src_col)
+
+  skill = random_parser.add_argument_group("Skill information")
+  skill.add_argument(
+      '--quantity', '-n',
+      type=int,
+      help='number of entries to generate',
+      required=True)
+  skill.add_argument(
+      '--skill-name', '-s',
+      type=str,
+      help='skill name to generate entries for',
+      required=True)
+  skill.add_argument(
+      '--skill-args', '-a',
+      type=str,
+      nargs='+',
+      action='append',
+      help=textwrap.dedent('''skill arguments. usage -a <arg-name> <val1> <val2> ...
+                              where val<i> are the possible values of the argument that will be chosen from at random
+                              * (placeholder value) if no values are given
+                           '''))
+
+  gauss = random_parser.add_argument_group("Mixed gaussian distribution")
+  gauss.add_argument(
+      '--gauss-params', '-g',
+      type=float,
+      help='mean and standard deviation (in that order)',
+      nargs='+',
+      required=True,
+      action='append')
```

Comment:
I've found it a bit difficult to pass parameters correctly. Changing to `nargs=2` would already help, but I would even suggest splitting this into `--gauss-mean` and `--gauss-stdev`

Reply:
i definitely agree on the `nargs=2` comment. I am not in favor of splitting it up further, personally as i would find it more cumbersome to then make sure that i have the right amount of means and stddevs defined. Instead, i improved the help text a bit to make the usage more clear.

Reply:
OK makes sense, thanks!